index.js

```diff
@@ -532,9 +558,17 @@
 
   //  Number$prototype$equals :: Number ~> Number -> Boolean
   function Number$prototype$equals(other) {
-    return typeof this === 'object' ?
-      equals(this.valueOf(), other.valueOf()) :
-      isNaN(this) && isNaN(other) || this === other && 1 / this === 1 / other;
+    return typeof other === 'object' ?
+      typeof this === 'object' &&
+        equals(this.valueOf(), other.valueOf()) :
+      isNaN(other) && isNaN(this) || this === other;
+  }
+
+  //  Number$prototype$lte :: Number ~> Number -> Boolean
+  function Number$prototype$lte(other) {
+    return isNaN(this) ? isNaN(other) :
+      (typeof this === 'object' ? this.valueOf() : this)
+        <= (typeof other === 'object' ? other.valueOf() : other);
   }
```

Comment:
The approach I suggested in https://github.com/sanctuary-js/sanctuary-type-classes/pull/46#discussion_r110643437 still seems best to me: provide definitions of `equals` and `lte` for both `NumberPrimitive` and `NumberObject`. This avoids arbitrarily deciding whether `0` comes before or after `new Number(0)`, because these values are incomparable.